rfcs/0000-question-sets.md

```diff
@@ -0,0 +1,150 @@
+# Summary
+
+A mechanism for allowing sets of questions to be published and used by multiple courses.
+
+# Basic Example
+
+A course can be designated as a question set by setting the `questionSet` flag in its `infoCourse.json` file:
+
+```json
+{
+  "uuid": "754dac56-fc1f-4888-a74a-1b365fd48376",
+  "questionSet": true
+}
+```
+
+Question sets will have a unique, human readable name, like `uiuc-data-structures`.
+
+Questions in a question set are defined the same way that they are for courses. That is, a directory in a `questions` directory. For example, the file `questions/avl-tree/info.json` defines a question with the ID `avl-tree`.
+
+Questions from a question set can be referenced with a combination of question set ID and question ID. For instance, in the following simplified `infoAssessment.json`, `uiuc-data-structures/avl-tree` references the `avl-tree` question from the `uiuc-data-structures` question set.
```

Comment:
We frequently have requests for #2360. To enable that, perhaps we should use some special character to indicate source course questions? Like `@uiuc-data-structures/avl-tree`?

Reply:
Really good point - agreed! We can prevent people from syncing questions that start with `@` to prevent any ambiguity.

Reply:
Yes! I was thinking earlier that we might want to start enforcing some rules about allowed characters in names. I noticed that CS 233 has some exams this semester that start with `@` (they are called `@Home1`, etc).